images: add fedora-42

allisonkarlitskaya opened the pull request:
* [x] image-refresh fedora-42

allisonkarlitskaya commented:
The bots had a mid-air collision.  I thought we got rid of that problem...

good run is here: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-fedora-42-43b1d133-20250210-131108/log.html
bad run is here: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-fedora-42-43b1d133-20250210-131111/log.html
complete statuses are here: https://api.github.com/repos/cockpit-project/bots/statuses/43b1d13352a90231d29a5e06d1d2f4583f2fa763 (in case we want to try to sort this out)

cockpituous commented:
image-refresh fedora-42 done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-42-20250210-132556

cockpituous commented:
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-fedora-42-43b1d133-20250210-131108/log.html

allisonkarlitskaya commented:
hm.  starter-kit is building and running the test (and it is functional, as per `TEST_SHOW_BROWSER=1`) but it's failing:

```
> warn: transport closed: disconnected
Traceback (most recent call last):
  File "/var/home/lis/src/starter-kit/main/test/check-application", line 19, in testBasic
    self.login_and_go("/starter-kit")
    ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^
  File "/var/home/lis/src/starter-kit/main/test/common/testlib.py", line 1931, in login_and_go
    self.browser.login_and_go(path, user=user, password=password, host=host, superuser=superuser,
    ~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                              urlroot=urlroot, tls=tls)
                              ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/home/lis/src/starter-kit/main/test/common/testlib.py", line 1001, in login_and_go
    self.enter_page(path.split("#")[0], host=host)
    ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/home/lis/src/starter-kit/main/test/common/testlib.py", line 909, in enter_page
    self.wait_not_visible(".curtains-ct")
    ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^
  File "/var/home/lis/src/starter-kit/main/test/common/testlib.py", line 841, in wait_not_visible
    self.wait_js_func('!ph_is_visible', selector)
    ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/home/lis/src/starter-kit/main/test/common/testlib.py", line 798, in wait_js_func
    self.wait_js_cond("%s(%s)" % (func, ','.join(map(jsquote, args))))
    ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/home/lis/src/starter-kit/main/test/common/testlib.py", line 795, in wait_js_cond
    raise Error(f"timeout\nwait_js_cond({cond}): {last_error.msg}") from None
testlib.Error: timeout
wait_js_cond(!ph_is_visible(".curtains-ct")): Error: .curtains-ct not found

# Result testBasic (__main__.TestApplication.testBasic) failed
# 1 TEST FAILED [36s on toolbx]
not ok 1 /var/home/lis/src/starter-kit/main/test/check-application TestApplication.testBasic > warn: transport closed: disconnected
Traceback (most recent call last):
  File "/var/home/lis/src/starter-kit/main/test/check-application", line 19, in testBasic
    self.login_and_go("/starter-kit")
    ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^
  File "/var/home/lis/src/starter-kit/main/test/common/testlib.py", line 1931, in login_and_go
    self.browser.login_and_go(path, user=user, password=password, host=host, superuser=superuser,
    ~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                              urlroot=urlroot, tls=tls)
                              ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/home/lis/src/starter-kit/main/test/common/testlib.py", line 1001, in login_and_go
    self.enter_page(path.split("#")[0], host=host)
    ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/home/lis/src/starter-kit/main/test/common/testlib.py", line 909, in enter_page
    self.wait_not_visible(".curtains-ct")
    ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^
  File "/var/home/lis/src/starter-kit/main/test/common/testlib.py", line 841, in wait_not_visible
    self.wait_js_func('!ph_is_visible', selector)
    ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/home/lis/src/starter-kit/main/test/common/testlib.py", line 798, in wait_js_func
    self.wait_js_cond("%s(%s)" % (func, ','.join(map(jsquote, args))))
    ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/home/lis/src/starter-kit/main/test/common/testlib.py", line 795, in wait_js_cond
    raise Error(f"timeout\nwait_js_cond({cond}): {last_error.msg}") from None
testlib.Error: timeout
wait_js_cond(!ph_is_visible(".curtains-ct")): Error: .curtains-ct not found

# Result testBasic (__main__.TestApplication.testBasic) failed
# 1 TEST FAILED [36s on toolbx]
not ok 1 /var/home/lis/src/starter-kit/main/test/check-application TestApplication.testBasic 
```

let's back out the testmap `main` addition for now.  We can clean this up a lot easier in a follow-up.

martinpitt commented:
@allisonkarlitskaya check https://github.com/cockpit-project/cockpit/commit/7957a125baada6a0b281bdddbbc32ea7cc256e95 that sounds related

allisonkarlitskaya commented:
> @allisonkarlitskaya check [cockpit-project/cockpit@7957a12](https://github.com/cockpit-project/cockpit/commit/7957a125baada6a0b281bdddbbc32ea7cc256e95) that sounds related

lol.  that'll happen if I'm testing an ancient starter-kit :)

allisonkarlitskaya commented:
```
1..1
qemu-img create -q -f qcow2 -b /var/home/lis/src/starter-kit/main/test/images/fedora-42 -F qcow2 /run/host/var/tmp/bots-run/cockpit-sn2c0jeb.qcow2
# 1 TEST PASSED [19s on toolbx]
ok 1 /var/home/lis/src/starter-kit/main/test/check-application TestApplication.testBasic [ND@0]

# TESTS PASSED [19s on toolbx, 0 destructive tests, 1 nondestructive tests: 0: 20s]
```

okay.  let's put the starter-kit main testmap update patch back, then.  oops!

allisonkarlitskaya commented:
Cool GitHub UI issue: it doesn't mention that I pushed anything because the thing that I pushed was already pushed previously and adding one commit back doesn't technically count as a force-push either.

Also: all the old statuses are back from before (but this part is easy to understand).

martinpitt reviewed (approved):
Nice, thanks! Naughty is already in place. I kicked the starter-kit run, but I trust you that it worked locally.

martinpitt commented:
Yeah, this doesn't run like that. We need to declare the _manual fedora-42 entries first and land that, but oh well.

martinpitt commented:
I sent #7419 to update testmap, and also run starter-kit/f42 there.
